src/cache_manager.cc

```diff
@@ -252,10 +227,13 @@ CacheManager::ParseHeaders(const HttpRequest * request, Mgr::ActionParams &param
     // TODO: use the authentication system decode to retrieve these details properly.
 
     /* base 64 _decoded_ user:passwd pair */
-    const auto basic_cookie(request->header.getAuthToken(Http::HdrType::AUTHORIZATION, "Basic"));
+    auto basic_cookie(request->header.getAuthToken(Http::HdrType::AUTHORIZATION, "Basic"));
 
-    if (basic_cookie.isEmpty())
-        return;
+    if (basic_cookie.isEmpty()) {
+        basic_cookie = request->header.getAuthToken(Http::HdrType::PROXY_AUTHORIZATION, "Basic");
```

Comment:
Let's clarify why we are adding the second getAuthToken() call:

```suggestion
        // squidclient does and other clients should use Proxy-Authorization to supply mgr credentials
        basic_cookie = request->header.getAuthToken(Http::HdrType::PROXY_AUTHORIZATION, "Basic");
```

We should also add a comment to explain the old/first call IMO:

```C++
// recognize Authorization header for backward compatibility
auto basic_cookie(request->header.getAuthToken(Http::HdrType::AUTHORIZATION, "Basic"));
```

if the above explanation is wrong, please use the right one, of course!

Reordering these calls/checks (i.e. checking what should be the default/primary option) makes sense, but has a potential to break that backward compatibility, so I would not do it in this PR.

Reply:
> squidclient does and other clients should use Proxy-Authorization to supply mgr credentials

This was true with the old cache_object scheme when requests were always directed to this Squid instance (i.e., were 'internal' requests), i.e., both Authorization and Proxy-Authorization could probably be used for the same purpose. Now, as we request 'some' cache manager with an http request, we cannot rely on Proxy-Authorization (dropped by the first proxy in chain). @yadij 's objection looks reasonable - we should also adjust squidclient to send Authorization header instead of Proxy-Authorization. @rousskov , do you have any objections?

Reply:
@eduard-bagdasaryan, I do not have any objections. I am glad your work exposed a bug in the official code! If Amos thinks that we should switch `mgr:` to origin `Authorization`, let's switch!

Disclaimer: I have not studied which authentication approach is the right one here. I am just helping polish/document your code changes and following Amos change requests.

Reply:
>  revert all these basic_cookie change

Done.